care/users/reset_password_views.py

```diff
@@ -114,176 +79,116 @@ class ResetPasswordConfirm(GenericAPIView):
             return Response(
                 {"detail": "Too Many Requests. Please try again later."},
                 status=status.HTTP_429_TOO_MANY_REQUESTS,
             )
 
-        # get token validation time
-        password_reset_token_validation_time = get_password_reset_token_expiry_time()
-
-        # find token
-        reset_password_token = ResetPasswordToken.objects.filter(key=token).first()
-
-        if reset_password_token is None:
-            return Response(
-                {"status": "notfound", "detail": "The password reset link is invalid"},
-                status=status.HTTP_404_NOT_FOUND,
+        # Verify token and get user
+        user, error_message = verify_password_reset_token(token)
+        if not user:
+            response = ResetPasswordResponse(
+                status="invalid", detail=error_message
+            ).model_dump()
+            return Response(response, status=status.HTTP_404_NOT_FOUND)
+
+        try:
+            validate_password(
+                password,
+                user=user,
+                password_validators=get_password_validators(
+                    settings.AUTH_PASSWORD_VALIDATORS
+                ),
             )
-
-        # check expiry date
-        expiry_date = reset_password_token.created_at + timedelta(
-            hours=password_reset_token_validation_time
-        )
-
-        if timezone.now() > expiry_date:
-            # delete expired token
-            reset_password_token.delete()
+        except Exception:
             return Response(
-                {"status": "expired", "detail": "The password reset link has expired"},
-                status=status.HTTP_404_NOT_FOUND,
-            )
-
-        # change users password (if we got to this code it means that the user is_active)
-        if reset_password_token.user.eligible_for_reset():
-            pre_password_reset.send(
-                sender=self.__class__, user=reset_password_token.user
-            )
-            try:
-                # validate the password against existing validators
-                validate_password(
-                    password,
-                    user=reset_password_token.user,
-                    password_validators=get_password_validators(
-                        settings.AUTH_PASSWORD_VALIDATORS
-                    ),
-                )
-            except ValidationError as e:
-                # raise a validation error for the serializer
-                raise exceptions.ValidationError({"password": e.messages}) from e
-
-            reset_password_token.user.set_password(password)
-            reset_password_token.user.save()
-            post_password_reset.send(
-                sender=self.__class__, user=reset_password_token.user
+                {
+                    "detail": "Password validation failed due to certain criteria not being met"
+                },
+                status=status.HTTP_400_BAD_REQUEST,
```

Comment:
```suggestion
        validate_password(
                password,
                user=user,
                password_validators=get_password_validators(
                    settings.AUTH_PASSWORD_VALIDATORS
                ),
)

```
validate_password will raise a validationerror with a correct message so no need to catch and return a response